### DISCUSSION TITLE
레벨3 런칭 페스티벌(4차 데모데이) 피드백

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#576] [기록해봐요]
# 피드백 RAW

1. 코드 기록 → 더 잘 쓰기 위해
설명 X ⇒ {”이 코드를 어디서 쓰나요?”, “언제 쓸 것 같으신가요?”, “~% 확률도 높아집니다.”}
생각을 불러 일으키는 문장!!
2. GPT로 목차나 요약 넣어주기
작성하기 귀찮으니 제목, 설명 자동으로
GPT로 찾아주기(검색 프롬프트)
3. GPT로 찾아주기(검색 프롬프트)
인출을 어떻게 할 것인가? → 70%는 이럴 때 씁니다 → 생각을 유도하다. 
4. 기록은 `인출`이 중요. 이 코드를 어디에 쓸 수 있는지.
ex) 아이디 유효성 → 써먹을 맥락이 설명에 들어가도록
인출은 언제 이걸 썼는지 메모하는 걸 자극한다. → 어디에 써먹을 것 같나요?
5. FE.conf, VSCode 익스텐션
6. 자동 Title 입력 기능 (정규식)
7. 태그엔 언어 정도는 자동으로, 간편할 듯. (함수명도?) → 백지에선 사용자가 인지부하가 온다. 
8. 취소하면 Alert 컨펌. 바로 `/`로 가고 날라감. 저장도. 
9. 400 뜨면 (업로드 실패 시 마이페이지로 넘어감) 서버에서 처리. 
10. 파일명 인풋이 잘 안 보임. SelectList Placeholder라도 있어야. 
11. 필수정보는 * 표시라도. 
12. 저장 버튼 위치
13. 마지막 소스코드는 지우지 못하게
14. 카테고리 멀리 가면 끊김
15. ~개씩 보기
16. 카테고리별 태그 → 클릭 시 태그 코드
17. 태그 소문자
18. 카테고리 변경 안 됨
19. 히스토리 다 쌓을 필요 없음. 업로드 7번 누르면 뒤로가기 7번 해야함
20. 이모지 등등 에러가 1자 이상으로 나옴, 에러 메시지 더 상세하게
21. 소스코드 길게 입력 안 되게 막기 (413 에러 ㅠㅠ)
22. 아이폰 폴더이름 공백으로 바꾸기 → 공백 입력됨 ㅠㅠ
23. 코드 복붙 검색할 수 있으면 좋을 것 같음 (유사도 검색)
24. 로그인을 했는데 ‘로그인 하러 가기’가 있어요
25. 여러 모듈로 나뉘는 템플릿을 서술식으로 관리 (ex. OAuth)
26. 아직 제 ID로 extension을 사용할 수 없어요! ‘-’
27. 랜딩페이지 로그인하러가기 안 보이게
28. 퍼블릭, 프라이빗 구분 있으면 좋겠다. 내 게시글을 다 모두에게 공유하고 싶지 않다
29. 링크, 출처도 입력되면 좋겠다. (익스텐션에서 자동으로 추출해주면 더 좋을듯)
30. 댓글기능 있으면 올린 사람에게 좋을 것 같다.
31. 헤더에 현재 페이지 반영해주기(색깔로)
32. 확장프로그램이 가장 궁금해요!! 그리고 한 템플릿에 여러 파일을 입력했을 때 나중에 찾기 쉬웠으면 좋겠어요. 
33. 필터링 - (그림), 리스트는 뱃지 클릭 가능, 개별항목은 크릭이 안 돼요~!
34. 템플릿 작성 - 설명란과 제목은 Text Area로 하면 좋을 것 같아요~
35. 인텔리제이 플러그인도 해주세요~
36. 파일명.js → 플레이스홀더 더 연하게 해줘 헷갈려
37. CPP의 import문 #으로 시작하는데 주석 처리 되는건지(파이썬처럼) 정상 적용되는 건지 좀 헷갈려요!!
38. 로그아웃하고 코드 수정버튼이 보여요. 하지만 업데이트는 안 되는듯!! 
39. 뒤로가기 제스처로 안 됨
40. 제목 작성을 까먹게 돼요 오류가 떠야 인지…?!
41. 코드 저작권 문제가 염려된다. 
42. 코드 다운로드 하고 싶다. 
43. 파일을 드래그 드롭으로 이동시킬 수 있으면 좋을 것 같아요
44. 포맷팅이 되면 더 좋겠다
45. 수정 삭제 버튼의 색상 비슷함. 동일하게 가거나 연필색 강조해줘
46. 코드 입력할 때 설명란의 placeholder가 설명을 꼭 써야 하는 것처럼 느껴졌어요. optional 인데
47. 로그인 했을 때 갑자기 템플릿 화면 떠서 당황했어요. 뭘 하라는 거지?
48. 어? 나는 닉네임 한 적 없는데 사람들은 한글 닉네임이네?
49. 메인 페이지에서 ‘구경가기’가 눈에 바로 안 들어와서 아쉬워요!
50. 삭제할 게 없는데 전체삭제 안 보여도 되지 않을까..?
51. 태그 이모지 x
52. 템플릿 제목 길이 제한
53. 템플릿 검색 조회 기능 (내꺼보다 남의 것을 보고 싶음)
54. 카테고리 길이 제한
55. 태그 몇 개 숨겨저 있는가? (+3)
56. 모든 태그 버튼 용도 모르겠음
57. 아이디 에러메시지 수정
58. 비밀번호 모든 특수문자 가능함
59. 카테고리 이름 개수 제한
60. 카테고리 편집 그냥 누르면 수정되게 해주세요. 
61. 특수문자만은 거르는게? 나을수도
62. 키워드 고정이었으면 좋겠어요.
63. JS 코드 보고싶을 때 자바스크립트, 자스, JS, js, Js 등등 바리에이션이 너무 많아서 혼란스러워요!
64. 713줄 이상 안 보임
65. 태그와 카테고리가 혼동된다.
66. 다른 사람 코드 볼 때 추천순(좋아요, 조회수 등)
67. 즐겨찾기 기능이 있으면 좋겠다.
68. 태그 중복 안 된다고 안 알려줘요.
69. 저장하고 다시 템플릿 목록을 보기 위해 ‘내 템플릿’을 누르는 게 불편해요. 
70. 저장 안 되는데 아무런 알림창이 안 떠요. (코드 많이 썼을 때)
71. 회원가입 시 한글, 특수문자 1자 이상 입력했는데 오류 뜸
72. 파일명 입력칸이 안 보여요. 
73. 내 정보
74. 코드 많이 쓰면 멈춤
75. 제스쳐로 뒤로 돌아가기 기능이 가능하면 좋겠다.
76. “템플릿”화면에서 이전 화면 이동 버튼이 있으면 좋겠어요!
77. “전체보기” 화면에서 검색 input 변경 때마다 fetch가 가서인지 반으이 조금 느린 것 같아요!
78. 코틀린 하이라이팅 지원해주세요!
79. 이전 로그인 사용자의 정보(태그목록)가 남아있는 경우 있음. 로그아웃 시 전부 날릴 것 ㅠㅠ
80. 구경가기에서 인기 많은 순(조회수, 좋아요?)으로 정렬해서 보고 싶어요. 
81. 단축키 안내
82. 익스텐션 저장 시 제목 입력
83. 템플릿 상세 페이지에서 태그 누르면 바로 검색되게
84. 태그에 콤마 추가 시 분리되어도 좋을 것 같아요.
85. 템플릿 복제도 좋을 것 같아요.
86. 아이디 특문 안 된다고 안 알려줌. 
87. 태그를 눌렀을 때 계속 이동해서 조~~금 보기가 힘들었어용
88. 동적인 인덴트 기능이 추가되었으면 좋겠어요. 자세한 내용은 진행자에게 전달했습니다. 
89. 다른 사람 코드도 검색으로 보고 싶어용
90. 태그 검색 ‘or’ 조건으로 해줘
91. 목록 조회에서 특정 템플릿 조회 후 목록조회로 돌아가면 페이지 리셋
92. 카테고리 모두 해제하는 버튼이 있으면 좋을 것 같아요~
93. 코드 작성 창에서 + 버튼이 더 크면 좋겠어요.
94. 다른 사람 코드에 ♥︎ 수로 검증된 코드인지 확인하고 싶어요
95. 새 템플릿 누르면 바로 커서가 제목쪽으로 가서 키보드로 제목을 바로 입력할 수 있으면 좋겠어요. 
96. 카테고리 편집 → 아이콘 보다 텍스트
97. 헤더에서 어느 위치인지 표시해달라. 
98. 구경가기에서 뭘 보고 싶을까. 관심사를 찾는 방법 → 검색
99. 여러 템플릿 → 원하는 템플릿을 찾아보세요
100. TOC 인지가 잘 안 된다. 
101. 제목, 설명은 다 입력하기 귀찮다. 주석으로 표현 가능? 제목에 대한 설명?
102. 책으로 치면 목차가 보이는게 좋지 않을까? 코드 썸네일보다는. 

# GPT: 분류

피드백을 관심사별로 분류하면 다음과 같습니다:

### 1. 사용성 및 UI/UX 관련 피드백

- **사용자 인터페이스(UI) 개선**:
[10, 12, 13, 15, 19, 22, 24, 31, 33, 36, 37, 38, 43, 45, 46, 49, 50, 56, 57, 60, 62, 64, 69, 72, 74, 75, 76, 77, 78, 79, 81, 85, 87, 92, 93, 95, 96, 97]
- **시각적 및 인터렉티브 요소 개선**:
[9, 14, 15, 18, 27, 34, 39, 47, 63, 73, 82]

### 2. 기능 개선 및 추가

- **자동화 및 개선 요청 기능**:
[1, 2, 6, 7, 8, 11, 16, 20, 23, 26, 30, 35, 42, 44, 52, 61, 66, 67, 68, 84, 88, 89, 90]
- **사용자 정보 및 접근성 관련 피드백**:
[24, 26, 27, 31, 32, 38, 79, 83, 80]

### 3. 버그 및 오류 수정 관련

- **기술적 이슈 및 오류 해결**:
[8, 20, 21, 39, 40, 41, 48, 57, 68, 70, 71, 74, 86]

### 4. 탐색 및 검색 기능

- **검색 및 분류 관련 기능**:
[3, 16, 23, 25, 53, 55, 63, 66, 68, 80, 83, 84, 90, 91]

### 5. 템플릿 관리 관련

- **템플릿 생성 및 관리**:
[28, 29, 35, 45, 54, 62, 76, 85, 86, 89, 92, 95, 100]

### 6. 태그 및 카테고리 관리

- **카테고리 및 태그 관리**:
[14, 18, 55, 56, 63, 64, 65, 66, 87, 90]

### 7. 커뮤니케이션 및 협업 기능

- **소셜 및 협업 기능 추가**:
[28, 30, 66, 67, 94]

### 8. 기타 제안

- **기타 제안사항**:
[41, 42, 43, 44, 78, 95, 96, 98, 101, 102]

# 피드백 데이터 사용 관련 고민

분류는 했는데, 위 데이터를 잘 사용하려면 어떻게 해야 할까? 떠오르는 아이디어는 다음과 같다. 

- 중복 피드백 정리
    - ex) 파일명 placeholder 가시성, 뒤로가기 제스처
- 엑셀로 저장
- 이슈로 등록할 수 있게 마크다운 형식으로 저장
- 우선순위 부여
    - 우선순위를 부여하고 팀원들에게 공유
        - 102개 항목을 모든 팀원이 검토할 필요 없다.
        - 내가 다 해서 회의를 빠르게 진행할 수 있다.

우선순위 부여가 가장 중요해 보인다. 우선순위가 높지 않다면 중복 제거의 필요성이나 파일화의 필요성도 높지 않다. 102개 항목을 하나씩 살펴보며 우선순위를 부여할 수 있을까? 

일단 해보자. 

우선순위를 정하는 기준을 먼저 생각해야 한다. 

1. 오류
    - 동작하지 않는 기능이나 오류가 발생하는 기능일수록 우선순위가 높다.
    - 개발자가 원하지 않는 방식으로 서비스가 동작하면 안 된다.
2. 빈도
    - 빈도가 높을 수록 우선순위가 높다.
    - 자주 언급되었다는 건 많은 사람들이 불편했다는 뜻이다.
3. 개선 제안
    - 사용자가 제안했다고 해서 반드시 반영해야 하는 것은 아니므로, 팀에서 다시 검토해야 한다.

자주 등장한 피드백을 먼저 추려내 보자. 

… 잘 안 된다. 무엇이 자주 등장한 것인가? 사람마다 단어가 다르다. 

그러면 일단 모든 번호에 키워드를 추려내 보자. 

1. 사용자 생각 유도
    - UI/UX 개선
2. GPT로 목차, 요약, 제목, 설명, 자동 입력
    - 새 기능
3. GPT로 찾아주기
    - 새 기능
4. 사용자 생각 유도
    - UI/UX 개선
5. 익스텐션
    - 새 기능
6. 자동 입력
    - 새 기능
7. 태그, 언어, 자동 입력
    - 새 기능
8. 취소 Alert
    - UI/UX 개선
9. 업로드 실패(400 Bad Request) 후 마이페이지
    - UI/UX 개선
10. 파일명 인풋 가시성
    - UI/UX 개선
11. 필수정보 * 표시
    - UI/UX 개선
12. 저장 버튼 위치
    - UI/UX 개선
13. 마지막 소스코드 삭제
    - UI/UX 개선
14. 카테고리 UI/UX
    - UI/UX 개선
15. *개씩 보기
    - 새 기능
16. 태그 클릭
    - 새 기능
17. 태그 소문자
    - UI/UX 개선
18. 카테고리 변경
    - 버그
19. 히스토리, 업로드, 뒤로가기
    - UI/UX 개선
20. 이모지, 1자 이상, 에러 메시지
    - 버그
21. 소스코드 입력 길이
    - UI/UX 개선
22. 아이폰, 폴더 이름 공백, 공백 입력
    - 버그
23. 코드 복사 붙여넣기 검색, 유사도 검색
    - 새 기능
24. 로그인 후 랜딩페이지
    - UI/UX 개선
25. 모듈, 템플릿, 서술식 관리
26. 익스텐션
    - 새 기능
27. 로그인 후 랜딩페이지
    - UI/UX 개선
28. Public, Private 공개 범위 구분
    - 새 기능
29. 링크, 출처 입력, 익스텐션, 자동 입력
    - 새 기능
30. 댓글
    - 새 기능
31. 헤더, 현재 페이지 반영
    - UI/UX 개선
32. 익스텐션
33. 태그  클릭
    - 새 기능
34. 템플릿 작성, 제목, 설명 칸을 Text Area로
    - UI/UX 개선
35. 인텔리제이 익스텐션
    - 새 기능
36. 파일명 placeholder
    - UI/UX 개선
37. Syntax highlighting
    - UI/UX 개선
38. 상세페이지 로그아웃 후 수정버튼
    - 버그
39. 뒤로가기 제스처
    - UI/UX 개선
40. 제목 작성 가시성
    - UI/UX 개선
41. 코드 저작권 염려
    - 기타
42. 코드 다운로드
    - UI/UX 개선
43. 파일 드래그 드롭
    - UI/UX 개선
44. 포맷팅
45. 수정 삭제 버튼 색상 가시성
    - UI/UX 개선
46. 설명 placeholder
    - UI/UX 개선
47. 로그인 후 첫 페이지
    - UI/UX 개선
48. id, username 혼동
    - UI/UX 개선
49. 구경가기 가시성
    - UI/UX 개선
50. 전체삭제 버튼
    - UI/UX 개선
51. 태그 이모지
    - UI/UX 개선
52. 템플릿 제목 길이 제한
    - UI/UX 개선
53. 템플릿 검색 조회
    - 새 기능
54. 카테고리 길이 제한
    - UI/UX 개선
55. 태그 숨김
    - UI/UX 개선
56. 모든 태그 버튼
    - UI/UX 개선
57. 

이것도 잘 안 된다. 완전히 정량적인 기준으로 평가하기에는 어려움이 있다. 피드백이 모두 주관적이고 정성적이기 때문이다. 

그렇다면 지금까지의 정보를 근거로 내 주관을 반영해 결정하는 게 최선으로 보인다. 

# 제우스 나름대로 선별한 우선순위 높은 피드백

아래의 기준을 고려했습니다.

1. 자주 언급되는지
2. 우리의 의도와 다르게 동작하는지
3. 투입해야하는 리소스가 상대적으로 적은지
4. 사용자 경험에 얼마나 도움이 되는지

3, 4번 기준은 제 주관을 반영했습니다.

## UI/UX

- [ ]  제목과 설명 입력 귀찮다, 입력을 인지하기 어렵다 (2, 6, 40, 95, 101)
- [ ]  파일명 입력 칸 안 보인다. (10, 36, 72)
- [ ]  로그인 후 랜딩페이지의 ‘로그인하러가기’ 안 보이게 (24, 27)
- [ ]  뒤로가기 제스처 (39, 75)
- [ ]  취소버튼 Alert (8)

## 새로운 기능

- [ ]  익스텐션 (5, 26, 29, 32)
- [ ]  구경가기 검색 (53, 89, 98)
- [ ]  조회수, 좋아요 (66, 94)
- [ ]  AI로 목차, 요약, 제목, 설명, 태그 언어 생성 (2, 6, 7, 101)

**Comment by jminkkk** [comment]
- 카테고리 여러개 한번에 생성하고 카테고리 목록 조회 시 시간 순 정렬이 아니다
- 쿠키 없는데 화면상으로는 로그아웃 상태가 아님 (로그아웃 아니길래 카테고리 삭제하려니까 그때서야 403)
